### PR TITLE
#161351147 Add pagination support for getting articles from a category

### PR DESCRIPTION
**Pull request by emmaadesile**
### What does this PR do?
- add pagination to articles that belong to a category
- return a default of 6 articles per page for category articles
- return a default of 6 articles per page for favorite articles

### Description of tasks to be completed
- Add pagination support to articles gotten from a category

### How can this be manually tested
After cloning the branch,
- run `npm install` to install dependencies
- login as an author and get the token
- use token to create articles
- add the articles to a category at this endpoint`POST /api/article-categories/:categoryName/articles` 
- get the articles from that category at this endpoint ` GET /api/article-categories/:categoryName/articles`

### What are the relevant pivotal tracker stories?
[Deilvers 161351147]

### CODE REVIEW

**Review by d-beloved — APPROVED** [review]
LGTM

File: server/controllers/categoryController.js [hunk 1]
```diff
@@ -257,34 +259,66 @@ class CategoryController {
...
+          offset
+        })
+          .then((articles) => {
+            if (articles.rows.length === 0) {
+              return res.status(404).json({
+                status: 'error',
+                error: 'There are no articles in this category'
+              });
+            }
+            const pagination = paginateArticle(articles, currentPage, limit);
```
Comment (CEOehis) [inline server/controllers/categoryController.js:307]: I don't think it should return an error if no articles exist for the category. It should return an empty array instead.

**Review by CEOehis — CHANGES_REQUESTED** [review]

File: server/controllers/categoryController.js (hunk 1 above)
Comment (emmaadesile) [reply server/controllers/categoryController.js:307]: I hav fixed it

Comment (emmaadesile) [reply server/controllers/categoryController.js:307]: I have fixed it

Comment (emmaadesile) [reply server/controllers/categoryController.js:307]: Thanks for the feedback. I have fixed it. It now returns an empty array for categories without articles.

Comment (emmaadesile) [reply server/controllers/categoryController.js:307]: Thanks for the feedback. I have fixed it.

Comment (emmaadesile) [reply server/controllers/categoryController.js:307]: Thanks for the feedback. I have implemented the feedback

Comment (emmaadesile) [reply server/controllers/categoryController.js:307]: Thanks for the feedback. I have implemented the feedback.

Comment (emmaadesile) [reply server/controllers/categoryController.js:307]: Thanks for the feedback. I have implemented the feedback.

Comment (emmaadesile) [reply server/controllers/categoryController.js:307]: Thanks for the feedback. I have implemented the feedback.

Comment (emmaadesile) [reply server/controllers/categoryController.js:307]: Thanks for the feedback. I have implemented the feedback.

Comment (emmaadesile) [reply server/controllers/categoryController.js:307]: Thanks for the feedback. I have implemented the feedback.

**Review by CEOehis — APPROVED** [review]
👍LGTM

File: server/controllers/categoryController.js [hunk 2]
```diff
@@ -257,34 +259,60 @@ class CategoryController {
    */
   static getAllArticlesForACategory(req, res, next) {
     const { categoryName } = req.params;
+    const { query } = req;
+    const limit = Number(query.limit) || 12;
```
Comment (waleCloud) [inline server/controllers/categoryController.js:263]: Great job! Your commit messages say a limit of 6 articles but your code is showing 12. Are my missing something?

**Review by waleCloud — CHANGES_REQUESTED** [review]
Hello @emmaadesile, Can we follow Andela standard of commit message limits in PRs?

File: server/controllers/categoryController.js (hunk 2 above)
Comment (emmaadesile) [reply server/controllers/categoryController.js:263]: @waleCloud  Thanks for spotting that. That must have been an oversight. Initially, the code was returning 6 articles per page, but after getting feedback from my teammates, I changed it to 12.

Comment (emmaadesile) [reply server/controllers/categoryController.js:263]: @waleCloud I have squashed the commits. I now have 2 commit messages

**Pull request merged**